libraries/shared/watcher/storage_watcher.go

```diff
@@ -198,7 +194,24 @@ func (watcher StorageWatcher) handleDiffWithInvalidHeaderHash(diff types.Persist
 		return fmt.Errorf(msg, diff.ID, maxBlockErr)
 	}
 	if diff.BlockHeight < int(maxBlock)-ReorgWindow {
```

Comment:
Wanted another set of eyes on this. When handling diffs with invalid header hash, it looks like if the diff's `blockHeight` is still within the ReorgWindow, then we're not changing the status on that diff. I think this makes sense, because it's possible that the given diff is actually canonical, but the reorg is currently in progress. So, I think by not setting the status, this diff will be pulled in again for processing in the future. Is that how others are understanding this as well?

Reply:
Yep 👍